src/Castle.Windsor/Core/Internal/ReflectionUtil.cs

```diff
@@ -150,14 +150,20 @@ public static Assembly GetAssemblyNamed(string assemblyName)
 
 		public static Assembly[] GetLoadedAssemblies()
 		{
-#if SILVERLIGHT
-			var list =
-				System.Windows.Deployment.Current.Parts.Select(
-					ap => System.Windows.Application.GetResourceStream(new Uri(ap.Source, UriKind.Relative))).Select(
-						stream => new System.Windows.AssemblyPart().Load(stream.Stream)).ToArray();
-			return list;
-#else
+#if FEATURE_APPDOMAIN
 			return AppDomain.CurrentDomain.GetAssemblies();
+#else
+			string basePath = AppContext.BaseDirectory;
+
+			var files = new DirectoryInfo(basePath).GetFiles("*.dll");
+			Assembly[] results = new Assembly[files.Length];
+			
+			for(int i = 0; i < files.Length; i++)
+			{
+				results[i] = System.Runtime.Loader.AssemblyLoadContext.Default.LoadFromAssemblyPath(files[i].FullName);
```

Comment:
Good to see `AppDomain.CurrentDomain.GetAssemblies()` comes back in .NET Standard 2.0, we should just require 2.0 😄 .

However as this code stands eagerly loading all DLLs from the bin directory will cause problems, firstly performance by forcing them to load and secondly exceptions on native DLLs in there (e.g. libcoreclr.dll).